### PR TITLE
MM-24293 Fix to have group channels return in quick switcher with space

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
  * Split the search term by spaces and use them to match for usernames in GM, if all search terms split by space matches users in GM then return the result 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24293

#### Screenshots
Untitled

### CODE REVIEW

**Review by esethna — APPROVED** [review]

**Comment by esethna** [comment]
Thanks @sudheerDev!

File: components/suggestion/switch_channel_provider.jsx [hunk 1]
```diff
@@ -277,6 +278,16 @@ function makeChannelSearchFilter(channelPrefix) {
...
         }
 
+        if (splitPrefixBySpace.length > 1) {
+            let match = true;
+            splitPrefixBySpace.forEach((searchPrefix) => {
+                if (!searchString.toLowerCase().includes(searchPrefix)) {
+                    match = false;
+                }
+            });
+            return match;
```
Comment (Willyfrog) [inline components/suggestion/switch_channel_provider.jsx:288]: Hopefully this is easier to read, feel free to ignore if you don't feel it is the case :)
```suggestion
            const lowerCaseSearch = searchString.toLowerCase();
            return splitPrefixBySpace.every((searchPrefix) => {
                return lowerCaseSearch.includes(searchPrefix);
            });
```

**Review by Willyfrog — APPROVED** [review]
Added a suggestion, but not really a blocker

**Review by deanwhillier — APPROVED** [review]
LGTM apart from @Willyfrog's suggestion.

File: components/suggestion/switch_channel_provider.jsx (hunk 1 above)
Comment (sudheerDev) [reply components/suggestion/switch_channel_provider.jsx:288]: That is nice, Thank you @Willyfrog

**Review by jgilliam17 — APPROVED** [review]
Thank you @sudheerDev 
Tested, looks good to merge.
- Verified channel switcher returns GM results when search terms used are split by space; GM results are returned when both search terms match users.

**Pull request merged**